crypto/params.c

```diff
@@ -518,7 +518,7 @@ int OSSL_PARAM_get_uint64(const OSSL_PARAM *p, uint64_t *val)
         switch (p->data_size) {
         case sizeof(double):
             d = *(const double *)p->data;
-            if (d >= 0 && d <= INT64_MAX && d == (uint64_t)d) {
+            if (d >= 0 && d == (uint64_t)d) {
```

Comment:
The original is wrong here too 👍 

Changing the comparison to `d >= 0 && d < 18446744073709551616.` should work.